test/dateTime.js

```diff
@@ -97,43 +93,103 @@ describe('dateTime', () => {
+					{timezone: 'America/Toronto', expectedGMTOffset: 4, month: 3, date: 8, hours: 7, minutes: 0},
+					{timezone: 'America/Toronto', expectedGMTOffset: 5, month: 3, date: 8, hours: 6, minutes: 59},
+					{timezone: 'America/Toronto', expectedGMTOffset: 4, month: 11, date: 1, hours: 5, minutes: 59},
+					{timezone: 'America/Toronto', expectedGMTOffset: 5, month: 11, date: 1, hours: 7, minutes: 0},
+					{timezone: 'America/Winnipeg', expectedGMTOffset: 5, month: 3, date: 8, hours: 8, minutes: 0},
+					{timezone: 'America/Winnipeg', expectedGMTOffset: 6, month: 3, date: 8, hours: 7, minutes: 59},
+					{timezone: 'America/Winnipeg', expectedGMTOffset: 5, month: 11, date: 1, hours: 6, minutes: 59},
+					{timezone: 'America/Winnipeg', expectedGMTOffset: 6, month: 11, date: 1, hours: 8, minutes: 0}
 				].forEach((test) => {
-					it(`${test.timezone}: should have expected GMT offset of ${test.expectedGMTOffset} hours on ${test.month}/${test.day} at ${test.hours}:${test.minutes} AM UTC `, () => {
-						const daylightSavingsDate = new Date(Date.UTC(2015, test.month - 1, test.day, test.hours, test.minutes));
+					it(`${test.timezone}: should have expected GMT offset of ${test.expectedGMTOffset} hours on ${test.month}/${test.date} at ${test.hours}:${test.minutes} AM UTC `, () => {
+						const UTCDateTime = {
+							month: test.month,
+							date: test.date,
+							year: 2015,
+							hours: test.hours,
+							minutes: test.minutes,
+							seconds: 0
+						};
 						documentLocaleSettings.timezone.identifier = test.timezone;
-						const result = convertUTCToLocalDateTime(daylightSavingsDate);
+						const result = convertUTCToLocalDateTime(UTCDateTime);
 						const expectedHour = test.hours - test.expectedGMTOffset;
-						const expectedMonth = test.month === 3 ? 'Mar' : 'Nov';
-						expect(result.toString()).to.contain(`${expectedMonth} 0${test.day} 2015 0${expectedHour}:${test.minutes}:00`);
+						const expected = {
+							month: test.month,
+							date: test.date,
+							year: 2015,
+							hours: expectedHour,
+							minutes: test.minutes,
+							seconds: 0
+						};
+						expect(result).to.deep.equal(expected);
 					});
 				});
 			});
 		});
 
 		describe('convertLocalToUTCDateTime', () => {
+			it('should return original date if timezone identifier is blank', () => {
+				const UTCDateTime = {
+					month: 8,
+					date: 25,
+					year: 2015,
+					hours: 12,
+					minutes: 28,
+					seconds: 0
+				};
+
+				documentLocaleSettings.timezone.identifier = '';
+				const result = convertLocalToUTCDateTime(UTCDateTime);
+				expect(result).to.deep.equal(UTCDateTime);
+			});
+
+			it('should throw if timezone identifier is invalid', () => {
+				const UTCDateTime = {
+					month: 8,
+					date: 25,
+					year: 2015,
+					hours: 12,
+					minutes: 28,
+					seconds: 0
+				};
+
+				documentLocaleSettings.timezone.identifier = 'FAKE';
+				expect(() => {
+					convertLocalToUTCDateTime(UTCDateTime);
```

Comment:
Is this the browser API that's throwing?

Reply:
yup